### PR TITLE
Several bootstrap and rocksdb improvements

### LINKED ISSUE #4865: Several bootstrap improvements
- when a client receives the Bootstrap error "all our changes are strictly after last_change_id, we can't be sure we did not miss any", we do not reset the state, and so the same slot is asked to the server (and the bootstrap fails in a loop). We should propagate the SlotTooOld error in the server instead
- Investigate and improve memory usage behaviour for bootstrap servers
- We recently increased the bootstrap limits, but we should also increase the default cache_history len
```
[ledger]
    # length of the changes history. Higher values allow bootstrapping nodes with slower connections
    final_history_length = 100
```
It is the max number of slots we allow between two bootstrap message.
With a default of 300s, it should be 600 slots instead of 100.

### PR DESCRIPTION
**Pull request by Leo-Besancon**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

Solves:
- Bad error handling (SlotTooOld error was not propagated correctly to client)
- Increase the default final_history_length to be consistent with the bootstrap default timeouts
- Avoid filling RocksDB read cache when iterating for bootstrap and other DB options
- Feature gate the use of jemalloc allocator (to be used to avoid memory fragmentation on linux)

### CODE REVIEW

File: massa-db-worker/src/massa_db.rs [hunk 1]
```diff
@@ -81,6 +81,13 @@ where
     ChangeIDSerializer: Serializer<ChangeID>,
     ChangeIDDeserializer: Deserializer<ChangeID>,
 {
+    /// If we read the db with these options, cache will be disabled (useful for scanning the whole db)
+    pub fn default_read_opts() -> ReadOptions {
```
Comment (Leo-Besancon) [inline massa-db-worker/src/massa_db.rs:85]: rename to read_opts_for_full_db_traversal

File: massa-db-worker/src/massa_db.rs [hunk 2]
```diff
@@ -621,27 +638,63 @@ impl RawMassaDB<Slot, SlotSerializer, SlotDeserializer> {
...
+
+        // Set hybrid bloom and ribbon filter, to reduce both memory and cpu usage, and add to cache
+        block_opts.set_hybrid_ribbon_filter(5.0, 2);
+        block_opts.set_cache_index_and_filter_blocks(true);
+        block_opts.set_pin_l0_filter_and_index_blocks_in_cache(true);
+        block_opts.set_pin_top_level_index_and_filter(true);
+        // Note: optimized it for memory depends on rocksdb feature "malloc-usable-size"
+        //block_opts.set_optimize_filters_for_memory(true);
+
+        db_opts.set_block_based_table_factory(&block_opts);
```
Comment (Leo-Besancon) [inline massa-db-worker/src/massa_db.rs:667]: Restrain to only the most useful ones (to solve the RAM issue). Otherwise, remove all of them.

File: massa-db-worker/src/massa_db.rs [hunk 3]
```diff
@@ -815,9 +868,9 @@ impl MassaDBController for RawMassaDB<Slot, SlotSerializer, SlotDeserializer> {
                 IteratorMode::From(key, Direction::Reverse)
             }
         };
-
+        let read_opts = Self::default_read_opts();
         Box::new(
-            db.iterator_cf(handle, rocksdb_mode)
+            db.iterator_cf_opt(handle, read_opts, rocksdb_mode)
```
Comment (Leo-Besancon) [inline massa-db-worker/src/massa_db.rs:873]: Check if this function is used for small db iteration, if we want to use the cache in some cases

File: massa-node/src/jemalloc_init.rs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+extern crate tikv_jemalloc_sys;
+
+#[used]
+static INIT: unsafe extern "C" fn(usize) -> *mut std::ffi::c_void = tikv_jemalloc_sys::malloc;
```
Comment (Leo-Besancon) [inline massa-node/src/jemalloc_init.rs:4]: We may want to test this out to better understand the impact, e.g.
- does this mean all allocations are done with Jemalloc?
- are there any edge-cases to consider?

Comment (Leo-Besancon) [reply massa-node/src/jemalloc_init.rs:4]: @damip By using the "stats" feature of tikv_jemalloc_sys, and printing the allocation stats, I can confirm jemalloc is used for all allocations of the binary, not just RocksDB allocations.
In my case, I printed stats in the main before and after allocating a huge Vec<u8>.

Comment (damip) [reply massa-node/src/jemalloc_init.rs:4]: this is quite surprising to me but ok x)

**Review by modship — APPROVED** [review]
